wordcount.py

```diff
@@ -37,37 +37,77 @@
 Optional: define a helper function to avoid code duplication inside
 print_words() and print_top().
 
+
 """
 
+#Time Spent: 4 hrs with google help and study group assists: Chris Wilson/Sean Bailey! 
+#Supported articles: 
+#https://github.com/mlafeldt/google-python-class/blob/master/basic/solution/wordcount.py
+
 import sys
 
-# +++your code here+++
-# Define print_words(filename) and print_top(filename) functions.
-# You could write a helper utility function that reads a file
-# and builds and returns a word/count dict for it.
-# Then print_words() and print_top() can just call the utility function.
+def word_count_dict(filename):
+  """Returns a word/count dict for this filename."""
+  # Utility used by count() and Topcount().
+  word_count = {}  # Map each word to its count
+  input_file = open(filename, 'r')
```

Comment:
This should be what's called a context manager,
with open <FILE> as f:
    ...do stuff...

That will automatically open and close the file.